waveorder/io/reader.py

```diff
@@ -0,0 +1,208 @@
+
+    def extract_data(self, master_ome):
+        """
+        extract all series from ome-tiff and place into dict of (pos: zarr)
+        :param master_ome: full path to master OME-tiff
+        :return:
+        """
+        log.info(f"extracting data from {master_ome}")
+        with TiffFile(master_ome) as tif:
+            for idx, tiffpageseries in enumerate(tif.series):
+                self._positions[idx] = zarr.open(tiffpageseries.aszarr(), mode='r')
+
+    def _get_master_ome_tiff(self, folder_):
+        """
+        search all ome.tif in directory
+        check the omexml for element tag corresponding to companion files (not-master-ome)
+
+        :param folder_: full path to folder containing images
+        :return: full path to master-ome tiff
+        """
+
+        from xml.etree import ElementTree as etree  # delayed import
+
+        def tag_search(root_, tag_name='BinaryOnly'):
+            """
+            returns True if tag_name is present
+            """
+            for element in root_:
+                if element.tag.endswith(tag_name):
+                    log.warning(f'OME series: not an ome-tiff master file')
+                    return True
+            return False
+
+        for file in os.listdir(folder_):
+            log.info(f"checking {file} for ome-master records")
+            if not file.endswith('.ome.tif'):
+                continue
+            with TiffFile(os.path.join(folder_, file)) as tiff:
+                omexml = tiff.pages[0].description
+                # get omexml root from first page
+                try:
+                    root = etree.fromstring(omexml)
+                except etree.ParseError as exc:
+                    try:
+                        omexml = omexml.decode(errors='ignore').encode()
+                        root = etree.fromstring(omexml)
+                    except Exception as ex:
+                        log.error(f"Exception while parsing root from omexml: {ex}")
+
+                # search for tag corresponding to non-ome-tiff-master files
+                if not tag_search(root, "BinaryOnly"):
+                    ome_master = file
+                    break
+                else:
+                    continue
+
+        return os.path.join(folder_, ome_master)
+
+    def get_zarr(self, position):
+        """
+        return a zarr array for a given position
+        :param position: int
+            position (aka ome-tiff scene)
+        :return: zarr.array
+        """
+        if not self._positions:
+            self.extract_data(self.master_ome_tiff)
+        return self._positions[position]
+
+    def get_array(self, position):
+        """
+        return a numpy array for a given position
+        :param position: int
+            position (aka ome-tiff scene)
+        :return: np.ndarray
+        """
+        if not self._positions:
+            self.extract_data(self.master_ome_tiff)
+        return np.array(self._positions[position])
```

Comment:
This can be simplified to
```
return np.array(self.get_zarr(position))
```

Reply:
i think they're equivalent.  This way there are fewer class function calls